warehouse/email/ses/tasks.py

```diff
@@ -0,0 +1,27 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import datetime
+
+from warehouse import tasks
+from warehouse.email.ses.models import EmailMessage
+
+
+CLEANUP_AFTER = datetime.timedelta(days=14)
+
+
+@tasks.task(ignore_result=True, acks_late=True)
+def cleanup(request):
```

Comment:
I think it might behoove us to cleanup unsuccessful deliveries on a slower pace than successful deliveries. I'd suggest we keep 30-90days of failures for debugging as it's quite possible for someone to not notice that things have gone awry in the short term. Particularly for email notifications for things that they did not initiate.

The alternative to holding onto logs for longer period is to persist some kind of summary as to _why_ the email address was unverified, rather than just that it was.

Either option works for me, but the latter actually has the benefit of persisting for a much longer period than even 30-90 days which is nice.

When a user finally reaches out, we can at least tell them what the "last straw" for their previously verified address was.

Reply:
To be clear, the boolean fields stored on accounts_email are probably enough to continue forward but may not capture all the necessary information to help a user get things back up and running. Also we may need more insight for certain spam greylisting/blacklisting situations.

Reply:
Set it up so failures are kept for 90 days, and success are kept for 14 days.